PatternPal/PatternPal.Extension/Views/StepByStepInstructionsView.xaml.cs

```diff
@@ -104,67 +107,92 @@ private void InitializeViewModelAndButtons()
-            CheckImplementationButton.IsEnabled = !_viewModel.CurrentInstruction.ShowFileSelector || ClassSelection.SelectedItem != null;
-
+            if (_viewModel.SBScontinue)
+            {
+                CheckImplementationButton.IsEnabled = ClassSelection.SelectedItem != null;
+            }
+            else
+            {
+                CheckImplementationButton.IsEnabled = 
+                    _viewModel.CurrentInstruction.FileId != "" || !_viewModel.SBScontinue;
+            }
+            
             NextInstructionButton.IsEnabled = false;
             ExpanderResults.ResultsView.ItemsSource = new List< PatternResultViewModel >();
         }
 
+        /// <summary>
+        /// Given the current step checks whether the implementation 
+        /// </summary>
         private void CheckImplementationButton_OnClick(
             object sender,
             RoutedEventArgs e)
         {
             NextInstructionButton.IsEnabled = false;
 
-            CheckInstructionRequest request = new CheckInstructionRequest
-                                              {
-                                                  InstructionSetName = _viewModel.InstructionSet.Name,
-                                                  InstructionId = _viewModel.CurrentInstructionNumber - 1,
-                                                  SelectedItem = _viewModel.SelectedcbItem,
-                                              };
-
-            LoadProject();
-
-            foreach (Project project in Projects)
+            if (_viewModel.SBScontinue)
             {
-                foreach (Document document in project.Documents)
+                CheckInstructionRequest request = new CheckInstructionRequest
                 {
-                    request.Documents.Add(document.FilePath);
-                }
-            }
-
-            try
-            {
-                RecognizeResult result = GrpcHelper.StepByStepClient.CheckInstruction(request).Result;
-
-                List< PatternResultViewModel > viewModels = new List< PatternResultViewModel >
-                                                            {
-                                                                new PatternResultViewModel(result)
-                                                                {
-                                                                    Expanded = true
-                                                                }
-                                                            };
+                    InstructionId = _viewModel.CurrentInstructionNumber - 1,
+                    SelectedItem = _viewModel.SelectedcbItem,
+                };
 
-                bool correct = result.Results.All(c => c.FeedbackType == CheckResult.Types.FeedbackType.FeedbackCorrect);
+                LoadProject();
 
-                ExpanderResults.ResultsView.ItemsSource = viewModels;
+                foreach (Project project in Projects)
```

Comment:
I'd to the same thing as the `RecognizerService` here, just pass either the path to a `.cs` file or the path to a `.csproj` file to the background service, you can get the files there, instead of keeping this logic in the extension.

Reply:
The logic regarding obtaining projects in the ide was replaced by having users provide a filepath(s) regardless. These filepaths are always known and sent to the service in a request.